### ISSUE TITLE
MRI Image Analysis - Deep Learning ResNet-50 (98% Train Accuracy)

### ISSUE THREAD

**Issue by rakheshkrishna2005** [UppuluriKalyani/ML-Nexus#695]
# 🧠 Alzheimer’s Disease Classifier - MRI Image Analysis

## 📝 Description

- Alzheimer’s Disease Classifier is a **computer vision application** that uses a **ResNet-50 model** to classify MRI images for Alzheimer's stages.
- Built with **Streamlit** for a user-friendly web interface, the app can classify multiple uploaded MRI images at once.
- Provides **insights** and **treatment recommendations** based on classification.

## 🎥 Demo Video
[Demo Video Link](https://github.com/user-attachments/assets/f86291be-cbab-40de-89b9-0426c4df6293)

## 🚀 Features

- 🖼️ **Multiple Image Upload** with easy preview option
- 📊 Detailed **classification report** for each image
- 🔍 **Four-class classification** for Alzheimer's stages
- 💡 Comprehensive **treatment recommendations** for each stage

## 💻 Tech Stack

- **Programming Language:** Python 🐍
- **Web Framework:** Streamlit 🌐
- **Deep Learning Framework:** PyTorch 🔥
- **Leveraged Pre-trained Model:** ResNet-50 🦾

Epoch 1/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [32:14<00:00, 12.02s/it]
Train Loss: 0.2492 Acc: 0.9035
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [00:45<00:00,  1.14s/it]
Validation Loss: 1.1234 Acc: 0.6357

Epoch 2/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [34:12<00:00, 12.75s/it]
Train Loss: 0.1134 Acc: 0.9609
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [01:24<00:00,  2.11s/it]
Validation Loss: 1.1876 Acc: 0.6747

Epoch 3/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [33:41<00:00, 12.56s/it]
Train Loss: 0.0739 Acc: 0.9756
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [01:27<00:00,  2.19s/it]
Validation Loss: 1.1369 Acc: 0.7185

Epoch 4/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [32:56<00:00, 12.28s/it]
Train Loss: 0.0349 Acc: 0.9893
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [01:26<00:00,  2.17s/it]
Validation Loss: 1.1831 Acc: 0.7209

Epoch 5/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [32:38<00:00, 12.16s/it]
Train Loss: 0.0685 Acc: 0.9777
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [01:24<00:00,  2.12s/it]
Validation Loss: 1.1935 Acc: 0.7287

Epoch 6/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [32:44<00:00, 12.20s/it]
Train Loss: 0.0476 Acc: 0.9844
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [01:28<00:00,  2.22s/it]
Validation Loss: 1.0082 Acc: 0.7365

Epoch 7/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [31:46<00:00, 11.84s/it]
Train Loss: 0.0389 Acc: 0.9883
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [00:42<00:00,  1.06s/it]
Validation Loss: 1.3168 Acc: 0.7209

Epoch 8/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [31:58<00:00, 11.91s/it]
Train Loss: 0.0395 Acc: 0.9869
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [00:33<00:00,  1.20it/s]
Validation Loss: 1.4977 Acc: 0.6951

Epoch 9/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [31:31<00:00, 11.75s/it]
Train Loss: 0.0158 Acc: 0.9957
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [00:33<00:00,  1.20it/s]
Validation Loss: 1.5907 Acc: 0.6841

Epoch 10/10
----------
Training: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 161/161 [43:32<00:00, 16.22s/it]
Train Loss: 0.0359 Acc: 0.9891
Validating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 40/40 [00:33<00:00,  1.19it/s]
Validation Loss: 1.3896 Acc: 0.7303
